ros/src/computing/planning/motion/packages/waypoint_maker/nodes/waypoint_marker_publisher/waypoint_marker_publisher.cpp

```diff
@@ -261,6 +265,60 @@ void createGlobalLaneArrayMarker(std_msgs::ColorRGBA color, const autoware_msgs:
 
 }
 
+void createLocalVelocityBarGraphMarker(const autoware_msgs::lane &lane_waypoint)
+{
+  visualization_msgs::Marker velocity_bar_graph_marker;
+  velocity_bar_graph_marker.header.frame_id = "map";
+  velocity_bar_graph_marker.header.stamp = ros::Time();
+  velocity_bar_graph_marker.ns = "local_velocity_bar_graph_marker";
+  velocity_bar_graph_marker.type = visualization_msgs::Marker::CYLINDER;
+  velocity_bar_graph_marker.action = visualization_msgs::Marker::ADD;
+  velocity_bar_graph_marker.scale.x = 0.2;
+  velocity_bar_graph_marker.scale.y = 0.2;
+  velocity_bar_graph_marker.color.a = g_graph_color[3];
+  velocity_bar_graph_marker.color.r = g_graph_color[0];
+  velocity_bar_graph_marker.color.g = g_graph_color[1];
+  velocity_bar_graph_marker.color.b = g_graph_color[2];
+  velocity_bar_graph_marker.frame_locked = true;
+
+  unsigned int count = 0;
+  for (auto el : lane_waypoint.waypoints)
+  {
+    double bar_graph_height = g_graph_height_ratio * el.twist.twist.linear.x;
+    velocity_bar_graph_marker.id = count++;
+    velocity_bar_graph_marker.pose = el.pose.pose;
+    velocity_bar_graph_marker.pose.position.z += bar_graph_height / 2.0;
+    velocity_bar_graph_marker.scale.z = bar_graph_height;
+    if(velocity_bar_graph_marker.scale.z == 0.0)
+      velocity_bar_graph_marker.scale.z = DBL_MIN; //for visualize
```

Comment:
Thank you for your review.
When the the cylinder height is 0 or less, a warning occurs in RViz.
For this reason, The value was set with a minimum value that does not occur a warning.

Reply:
Ah, I see. But this is like just hacking, so you should not push that Marker object to MarkerArray. Plz modify.

Reply:
If you do not push a marker with speed 0 to the marker array, the old ID will remain on RViz.
For erasing the old ID, we need to think about another way.
As an alternative, we can manage display / non-display with the "action" argument of the marker.

Reply:
It is necessary to update every frame. If you do not update the old name space, it will  remain on RViz. Therefore, another method must be considered.
In order to make the cylinder height  be 0 or more, I propose a method of adding a minute amount which does not affect the display all the time.